runtime/caml/platform.h

```diff
@@ -67,10 +67,22 @@ Caml_inline void cpu_relax(void) {
 
 /* Atomic read-modify-write instructions, with full fences */
 
-Caml_inline uintnat atomic_fetch_add_verify_ge0(atomic_uintnat* p, intnat v) {
-  uintnat result = atomic_fetch_add(p,v);
-  CAMLassert ((intnat)result > 0);
-  return result;
+/* Decrements the `atomic_uintnat` indicated by `p`, checking for
+ * underflow. */
+
+Caml_inline void atomic_decrement_count(atomic_uintnat* p) {
+  uintnat old = atomic_fetch_sub(p, 1);
+  CAMLassert ((intnat)old > 0);
+  (void)old;
+}
+
+/* Increments the `atomic_uintnat` indicated by `p`, checking for
+ * overflow. */
+
+Caml_inline void atomic_increment_count(atomic_uintnat* p) {
+  uintnat old = atomic_fetch_add(p, 1);
+  CAMLassert ((intnat)(old+1) != 0);
+  (void)old;
```

Comment:
Aren't these more-or-less duplicates of functions in `caml/atomic_refcount.h`?

Reply:
I suppose they are, except for the CAMLassert, and they should probably go in `camlatomic.h` (`platform.h` seems like a strange choice for them). There's a bit of a dependency tangle there (too much stuff in `misc.h`) but I'll give it a go.

Reply:
This is now addressed in #13593. I'll rebase.